deploy/gcloud/gcloud_up.sh

```diff
@@ -0,0 +1,186 @@
+    > amppkg_consumer.toml
+fi
+
+if [ -f "amppkg_renewer.toml" ]; then
+  echo "amppkg_renewer.toml exists. Skipping generation."
+else
+  cat amppkg_renewer_template.toml \
+    | sed 's/$(AMP_PACKAGER_CERT_FILENAME)/'"$AMP_PACKAGER_CERT_FILENAME"'/g' \
+    | sed 's/$(AMP_PACKAGER_CSR_FILENAME)/'"$AMP_PACKAGER_CSR_FILENAME"'/g' \
+    | sed 's/$(AMP_PACKAGER_PRIV_KEY_FILENAME)/'"$AMP_PACKAGER_PRIV_KEY_FILENAME"'/g' \
+    | sed 's/$(AMP_PACKAGER_DOMAIN)/'"$AMP_PACKAGER_DOMAIN"'/g' \
+    | sed 's/$(ACME_EMAIL_ADDRESS)/'"$ACME_EMAIL_ADDRESS"'/g' \
+    | sed 's,$(ACME_DIRECTORY_URL),'"$ACME_DIRECTORY_URL"',g' \
+    > amppkg_renewer.toml
+fi
+
+# Generate the yaml files that have the project id and docker image version tag with proper values.
+if [ -f "amppackager_cert_renewer.yaml" ]; then
+  echo "amppackager_cert_renewer.toml exists. Skipping generation."
+else
+  echo "Generating renewer YAML file ..."
+  cat amppackager_cert_renewer_template.yaml | sed 's/$(PROJECT_ID)/'$PROJECT_ID'/g' | sed 's/$(AMP_PACKAGER_VERSION_TAG)/'$AMP_PACKAGER_VERSION_TAG'/g' > amppackager_cert_renewer.yaml
+fi
+
+if [ -f "amppackager_cert_consumer.yaml" ]; then
+  echo "amppackager_cert_consumer.toml exists. Skipping generation."
+else
+  echo "Generating consumer YAML file ..."
+  cat amppackager_cert_consumer_template.yaml | sed 's/$(PROJECT_ID)/'$PROJECT_ID'/g' | sed 's/$(AMP_PACKAGER_VERSION_TAG)/'$AMP_PACKAGER_VERSION_TAG'/g' > amppackager_cert_consumer.yaml
+fi
+
+# Build docker images for the Amppackager renewer and consumer, if necessary.
+# Renewer and consumer are the same binaries, passed different command line
+# arguments. If you don't want to rebuild the binaries, you can comment out the
+# next 2 lines as well as the docker images and push commands below.
+docker build -f Dockerfile.consumer -t gcr.io/${PROJECT_ID}/amppackager:${AMP_PACKAGER_VERSION_TAG} .
+docker build -f Dockerfile.renewer -t gcr.io/${PROJECT_ID}/amppackager_renewer:${AMP_PACKAGER_VERSION_TAG} .
+
+# To check that it succeeded, list the images that got built.
+docker images
+
+# https://cloud.google.com/container-registry/docs/advanced-authentication#gcloud-helper
+gcloud auth configure-docker
+
+# Push the docker image into the cloud container registry.
+# See: https://cloud.google.com/container-registry/docs/overview
+# See: https://cloud.google.com/container-registry/docs/pushing-and-pulling
+docker push gcr.io/${PROJECT_ID}/amppackager:${AMP_PACKAGER_VERSION_TAG}
+docker push gcr.io/${PROJECT_ID}/amppackager_renewer:${AMP_PACKAGER_VERSION_TAG}
+
+gcloud config set project $PROJECT_ID
+gcloud config set compute/zone $COMPUTE_ENGINE_ZONE
+
+# Allow 10 nodes maximum for this cluster.
+#gcloud container clusters create amppackager-cluster --tags allow-8080 --num-nodes=10
+echo "Creating kubernetes cluster"
+gcloud container clusters create amppackager-cluster --num-nodes=10 --enable-ip-alias --metadata disable-legacy-endpoints=true
```

Comment:
Naive question number 452: Are all servers in a kube cluster allowed to talk to each other & otherwise isolated from the outside world? Or do we need a bit of documentation on setting up the firewall somewhere?

Reply:
They are allowed to talk to each other and isolated from outside world without you explicitly exposing them.